_studio/mfx_lib/vpp/src/mfx_vpp_sw_internal.cpp

```diff
@@ -327,24 +327,20 @@ mfxStatus GetExternalFramesCount(VideoCORE* core,
 } // mfxStatus GetExternalFramesCount(...)
 
 // all check must be done before call
-mfxStatus GetCompositionEnabledStatus(mfxVideoParam* pParam )
+bool IsCompositionMode(mfxVideoParam* pParam)
 {
-    mfxU32 bufferIndex;
-
-    if( pParam->ExtParam && pParam->NumExtParam > 0 )
+    if (pParam->ExtParam && pParam->NumExtParam > 0)
     {
-        for( bufferIndex = 0; bufferIndex < pParam->NumExtParam; bufferIndex++ )
+        for (mfxU32 i = 0; i < pParam->NumExtParam; ++i)
```

Comment:
this is actually find_if (or in this particular case any_of):
```
MFX_CHECK(pParam && pParam->ExtParam && pParam->NumExtParam > 0, false);

return std::any_of(pParam->ExtParam, pParam->ExtParam + pParam->NumExtParam, [](const mfxExtBuffer* pBuffer){ return pBuffer && pBuffer->BufferId == mfxU32(MFX_EXTBUFF_VPP_COMPOSITE); })
```

But I don't really insist on such change, missing pointer check is real issue here